src/apps/service/apps/jobs/sharing.clj

```diff
@@ -150,42 +161,94 @@
   (process-job-inputs (partial share-input-file sharer sharee) apps-client job))
 
 (defn- share-job
-  [apps-client sharer sharee {job-id :analysis_id level :permission}]
-  (if-let [job (jp/get-job-by-id job-id)]
+  [update-fn apps-client sharer sharee {job-id :analysis_id level :permission}]
+  (let [job-id (uuidify job-id)
```

Comment:
The job ID here has been decoded from the retrieved asynchronous task, so it has to be converted to a UUID.